maze_transformer/evaluation/path_evals.py

```diff
@@ -97,4 +88,40 @@ def fraction_connections_adjacent(
     ) -> float:
         """fraction of connections in prediction which are are valid paths on the maze"""
 
-        return PathEvals.num_connections_adjacent(maze, prediction) / len(prediction)
+        num_connections = len(prediction) - 1
+        return PathEvals.num_connections_adjacent(maze, prediction) / num_connections
+
+    @register_method(evals)
+    @staticmethod
+    def exact_path_predicted(solution: MazePath, prediction: MazePath, **_) -> float:
+        """Was the maze successfully solved?"""
+        return float(np.all(solution == prediction))
```

Comment:
I would add a comment here explaining that `==` doesn't actually crash for arrays of different sizes -- I was certainly under the impression that they do

Reply:
This is actually a good catch! It does work (numpy broadcasts), but it is deprecated behaviour and emits a warning. Fixing.